### DISCUSSION TITLE
[vcpkg_extract_archive] 7-Zip Alternitives

### DISCUSSION THREAD

**Discussion by gerard-ryan-immersaview** [microsoft/vcpkg#40869] [Ideas]
### Is your feature request related to a problem? Please describe.

My company disallows the usage of [7-Zip](https://7-zip.org/).

### Proposed solution

currently, the only approved alternative is PeaZip which [does appear to have a CLI](https://peazip.github.io/peazip-command-line.html). So I'm wondering If it would be acceptable to replace 7-zip within `vcpkg_extract_archive` likely with a flag.

### Describe alternatives you've considered

My company may be able to implement this as a forked modification, but the preference would be to upstream this to help mitigate the maintenance of such a low-level feature.

**Comment by mroszkobt** [comment]
7-Zip is open source just like PeaZip. Is this just a discrimination because the 7-zip author is russian?

If I were talking security, the Peazip author doens't even use git correctly. They drag and drop bulk file updates in the github web interface.

https://github.com/peazip/PeaZip/commits/sources/

I would be more worried about PeaZip having things leak in that shouldn't.

**Comment by gerard-ryan-immersaview** [comment]
@mroszkobt the specifics of the decision have not been made clear to me, I've just been tasked with the work,  
I appreciate the insight into PeaZip Git practices though.

**Comment by Sewer56** [comment]
Potential casual racism/xenophobia from ImmersaView aside.

PeaZip is a UI wrapper around many precompiled binaries, including but not limited to `7z.dll`.

So you actually get an unmodified 7-Zip binary with PeaZip.

**Comment by gerard-ryan-immersaview** [comment]
Thanks for the feedback everyone, I've been instructed that we'll no longer be pursuing this course of work, so I'll close this discussion.
